### PR TITLE
feat(ui): new platformicons on dashboard

### PR DESCRIPTION
**Pull request by ckj**
This introduces a Platformicon component that renders our new platformicons on the projects tab. Here's how they look in context:

screen shot 2018-05-04 at 9 04 40 pm

And here are all the current icons in storybook:

screen shot 2018-05-05 at 9 43 42 pm

### CODE REVIEW

**Comment by billyvg** [comment]
I think it's because webpack is extracting svgs and it's causing a conflict since they are named the same.

File: src/sentry/static/sentry/app/components/platformicon.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  let icon = PLATFORM_TO_ICON[platform];
+
+  if (!icon) {
+    return 'generic';
+  } else {
+    return icon;
+  }
+}
+
+const Platformicon = ({platform, size, width, height, ...props}) => {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/platformicon.jsx]: nit: can we rename this to `platformIcon` (and the file to be `platformIcon`) to be consistent?

File: src/sentry/static/sentry/app/views/organizationDashboard/platformList.jsx [hunk 2]
```diff
@@ -18,8 +19,8 @@ class PlatformList extends React.Component {
 
   getIcon(platform) {
     return (
-      <StyledPlatformIconWrapper key={platform} className={platform}>
-        <StyledPlatformIcon className={`platformicon platformicon-${platform}`} />
+      <StyledPlatformIconWrapper key={platform}>
+        <StyledPlatformIcon platform={platform} size="24" className={platform} />
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDashboard/platformList.jsx]: Do we still need className here?

File: docs-ui/components/platformicon.stories.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import {storiesOf} from '@storybook/react';
+import {withInfo} from '@storybook/addon-info';
+
+import Platformicon from 'app/components/platformicon';
+
+const StyledPlatformicon = styled(Platformicon)`
+  margin: 0 15px 15px 0;
+`;
+
+storiesOf('Platformicon', module).add(
```
Comment (billyvg) [inline docs-ui/components/platformicon.stories.js]: 🎉 Need these for our other icons too

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/views/organizationDashboard/platformList.jsx (hunk 2 above)
Comment (ckj) [reply src/sentry/static/sentry/app/views/organizationDashboard/platformList.jsx]: Probably not, but I left it there so the test would still work

File: src/sentry/static/sentry/app/components/platformicon.jsx (hunk 1 above)
Comment (ckj) [reply src/sentry/static/sentry/app/components/platformicon.jsx]: The lib is 'platformicons' (all one word).

Comment (ckj) [reply src/sentry/static/sentry/app/components/platformicon.jsx]: Oh, but I see that we have StyledPlatformIcon, etc. I'll fix.

File: src/sentry/static/sentry/app/views/organizationDashboard/platformList.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationDashboard/platformList.jsx]: Let's remove it, we can update the test to search via props, e.g.

```javascript
    expect(platformList.find('StyledPlatformIcon[platform="javascript"]')).toHaveLength(1);
```

**Pull request merged**